README.md

```diff
@@ -1,13 +1,102 @@
+3. Once your repository is cloned, go to the repository's directory `[your_directory]/generative-art/` and install all the packages and dependencies for the website by typing (if you work on Mac it should be like `Users/[your_user]/[your_directory]/generative-art/`):
+
+   ```
+   $ npm install
+   ```
+
+4. Check if the packages are correctly installed by typing:
+
+   ```
+   npm run start
+   ```
+
+5. This should open a window in your browser with the site running at http://localhost:3000.
+
+## Before Submitting a PR
+
+<em>Please do this everytime you submit a PR</em>
+
+1. Sync your fork to keep it up-to-date with the upstream repository following the next commands or this [tutorial](https://help.github.com/articles/syncing-a-fork/). First fetch the upstream repo and its commits -commits will be stored on your local fork- and then merge changes from upstream to your local:
+
+   ```
+   $ git fetch upstream
+   $ git merge upstream/master
+   ```
+
+2. Check if your changes are correct and don't break the website render by typing `npm run start`.
+3. Commit the files you have changed, type:
+
+   ```
+   $ git add -A
+   $ git commit -m "add a message to your commit"
+   $ git push
+   ```
+
+4. Commit to your repository at your github account and create a new PR. Click the _Pull Request_ tab on your fork page and then click the green button _New Pull Request_.
+5. Before submitting the PR, make sure to run `npm run format` on your project-directory to make the code properly formatted.
+
+## PR Format
+
+_The Pull-Request should contain the following information._
+
+1. Fix: #issue_number
+2. Feature Added/Changed
+3. Screenshots of Change
+
+## File Structure
+
+In the repository, there are two main directories which we need to focus:
+
+```
+genertative-art/
+    src/
+    public/
+```
+
+- `src/` - All the different components that are present in the website
+
+  - `components/`- The Website is made using React.js Framework and this directory contains all the different React Components
+    - `ArtDisplay/` - This repository contains all the different P5.js sketches and the `ArtDisplay` Component which renders all the different sketches. **If you want to add a new art-piece to the project, a sample template is given in this directory for reference**.
+    - `Card/` - It contains the `Card` component used to display the information regarding all art-pieces.
+    - `Dashboard/` - It contains the `Dashboard` component which renders the homepage with all the different art-pieces.
+    - `App.js` - It renders all the different React components of the website.
+
+- `public/` - It contains the `data.json` file which contains the details regarding all the art-pieces. **For adding a new-art piece, the data related to it needs to be added to this file first.** This directory also contains `contributor-data.json` file which contains the information regarding the contributors to the project.
```

Comment:
I don't think we need to explain the React files. Just guidelines on how to add a new art piece would be sufficient.